### PR TITLE
[Refactor] support flip test for heatmap heads

### PR DESCRIPTION
**Pull request by ly015**
## Motivation

Support test time augmentation by flipping for heatmap heads (`HeatmapHead`, `CPMHead`, `MSPNHead` and `ViPNASHead`)

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/models/utils/tta.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        B, C, H, W = heatmaps.shape
+        assert len(flip_indices) * 3 == C
+        heatmaps = heatmaps.view(B, C // 3, 3, H, W)
+        heatmaps = heatmaps[:, flip_indices].flip(-1)
+        heatmaps[:, :, 1] = -heatmaps[:, :, 1]
+        heatmaps = heatmaps.view(B, C, H, W)
+    else:
+        raise ValueError(f'Invalid flip_mode value "{flip_mode}"')
+
+    if shift_heatmap:
```
Comment (Tau-J) [inline mmpose/models/utils/tta.py:45]: Could provide any corresponding reference in the previous codes? I'm not sure whether it is appropriate to shift heatmaps after udp_combined results

Comment (ly015) [reply mmpose/models/utils/tta.py:45]: Yes, `shift_heatmap` is not used for any UDP model.

Comment (Tau-J) [reply mmpose/models/utils/tta.py:45]: How about setting a limit on it in case of misusing?

Comment (ly015) [reply mmpose/models/utils/tta.py:45]: Doesn't seem easy to do so. The function `flip_heatmap` and even the head are not supposed to be able to directly check the codec type. It won't be a problem, I think. In most cases, users will follow our configs on these settings.

**Review by Tau-J — APPROVED** [review]

**Comment by liqikai9** [comment]
I have modified the example config `configs/tests/hrnet_w32_coco_256x192.py`. And now we can run the command like:
`python tools/test.py configs/tests/hrnet_w32_coco_256x192.py ~/checkpoints/hrnet_w32_coco_256x192-c78dce93_20200708.pth`
or 
`./tools/slurm_test.sh mm_dev test configs/tests/hrnet_w32_coco_256x192.py ~/checkpoints/hrnet_w32_coco_256x192-c78dce93_20200708.pth`
to test the config with one GPU or multiple GPUs. 

I modify these scripts' mode: `tools/*.sh` to make them executable, changing their modes from 100644 to 100755. 

The mAP results are aligned with these items listed here: `https://github.com/open-mmlab/mmpose/blob/master/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_coco.md`, with about 1e-4 difference with the results in `hrnet_coco.md`.

**Review by liqikai9 — APPROVED** [review]

**Comment by ly015** [comment]
Changing file modes seems unnecessary. You can always execute them by `sh xxx` or `python xxx`.

**Pull request merged**